src/reporters/index.ts

```diff
@@ -1,228 +1,52 @@
-import { Hash } from "@polkadot/types/interfaces/runtime";
-import { GenericExtrinsic, GenericEvent } from "@polkadot/types/";
-import { Codec } from "@polkadot/types-codec/types";
-import { logger } from "../logger";
-import * as nodemailer from "nodemailer";
-import * as openpgp from 'openpgp';
-import * as sdk from "matrix-js-sdk";
-import { appendFileSync } from "fs";
-import { readFileSync } from 'fs'
-import { EmailConfig, ExtendedAccount, FsConfig, MatrixConfig, ReportType } from "..";
-import { ApiPromise } from "@polkadot/api";
+import { Report } from "../chain_monitor";
 
-const MAX_FORMATTED_MSG_LEN = 256;
-
-enum COLOR {
-	Primary = "#a3e4d7",
-}
-
-export function methodOf(input: ReportInput): string {
-	if (input.type === ReportType.Event) {
-		return input.inner.method.toString()
-	} else {
-		return (input.inner as GenericExtrinsic).meta.name.toString()
-	}
-}
-
-interface ReportInput {
-	account: ExtendedAccount,
-	type: ReportType,
-	inner: GenericEvent | GenericExtrinsic;
-}
-
-export interface Report {
-	hash: Hash,
-	number: number,
-	chain: string,
-	timestamp: number,
-	inputs: ReportInput[]
-	api: ApiPromise,
-}
+export { MatrixReporter } from "./matrix";
+export { EmailReporter } from "./email";
+export { ConsoleReporter } from "./console";
+export type { Report } from "../chain_monitor";
 
 export interface Reporter {
-	report(report: Report): Promise<void>;
-
-}
-
-export class GenericReporter  {
```

Comment:
Why is this removed, and replaced with a function? 

Again, you are removing useful programming abstractions. If your goal is simplicity, I disagree that this helps. 

Now you are hard-coding things like "how to get a subscan link" inside of the `ReportHTML`. What if in a month, we need to add a markdown exporter as well? should we hard-code this in there as well? 

This is exactly why I abstracted this in `GenericReporter`.